recipes-ni/ni-systemreplication/files/nisystemreplication

```diff
@@ -100,15 +106,27 @@ image_get () {
 
 # ---------------------- Set Image ----------------------
 
-# Presents a list of existing images and let's user select one
+# Presents a list of existing compatible images and let's user select one
 select_image_from_list () {
 	[ ! -d $NIRECOVERY_MOUNTPOINT/Images ] && return
 
-	readarray -t images <<< "$(find $NIRECOVERY_MOUNTPOINT/Images -name systemimage.tgz | sort | sed "s,$NIRECOVERY_MOUNTPOINT/Images/,," | sed 's,/systemimage.tgz,,')"
+	images=()
+
+	for image in $(ls $NIRECOVERY_MOUNTPOINT/Images); do
+		if [ ! -f $NIRECOVERY_MOUNTPOINT/Images/$image/systemimage.tgz ] || [ ! -f $NIRECOVERY_MOUNTPOINT/Images/$image/systemimage.conf ]; then
+			continue
+		fi
+
+		local ImageDeviceDesc=$(sed 's/DeviceDesc=//g' $NIRECOVERY_MOUNTPOINT/Images/$image/systemimage.conf)
```

Comment:
Probably `awk` is better for pulling a key-value pair out of the conf file; right? What happens if the conf file contains other values?

Reply:
> Probably awk is better for pulling a key-value pair out of the conf file; right?
Are you thinking something like `awk -F= '{if ($1 == "DeviceDesc") print $2}'`?

> What happens if the conf file contains other values?
We could let that be undefined behavior as that could only happen when user has tampered with the file.

Reply:
Updated to `awk`